src/lib.rs

```diff
@@ -51,43 +52,77 @@ fn compile(
     format: Option<&str>,
     ppi: Option<f32>,
 ) -> PyResult<PyObject> {
-    let input = input.canonicalize()?;
-    let root = if let Some(root) = root {
-        root.canonicalize()?
-    } else if let Some(dir) = input.parent() {
-        dir.into()
-    } else {
-        PathBuf::new()
-    };
-    let resource_path = Python::with_gil(|py| resources_path(py, "typst"))?;
+    // static world for each thread
+    thread_local! {
+        static WORLD: RefCell<Option<SystemWorld>> = RefCell::new(None);
+    }
 
     py.allow_threads(move || {
-        // Create the world that serves sources, fonts and files.
-        let mut default_fonts = Vec::new();
-        for entry in walkdir::WalkDir::new(resource_path.join("fonts")) {
-            let path = entry
-                .map_err(|err| PyIOError::new_err(err.to_string()))?
-                .into_path();
-            let Some(extension) = path.extension() else {
-                continue;
-            };
-            if extension == "ttf" || extension == "otf" {
-                default_fonts.push(path);
+        // canonicalize the input path and root path
+        let input = input.canonicalize()?;
+        let root = if let Some(root) = root {
+            root.canonicalize()?
+        } else if let Some(dir) = input.parent() {
+            dir.into()
+        } else {
+            PathBuf::new()
+        };
+        // if the world is not initialized or the input path, root path is changed,
+        // reinitialize the world
+        if WORLD.with(|world| {
```

Comment:
What if `font_paths` changed?

Reply:
Thanks for your reminder, i have added the code detecting if the font paths is changed.